```
    0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:06 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:07 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:08 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:09 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:10 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:11 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:12 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:13 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:14 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:15 --:--:--     0
    0     0    0     0    0     0      0      0 --:--:--  0:00:15 --:--:--     0
  curl: (7) Failed to connect to trustworthy.systems port 443 after 15667 ms: Couldn't connect to server
  make[1]: *** [Makefile:91: 85000f3f42a882e4476e57003d53f2bbec8262b0-linux] Error 7
  make[1]: Leaving directory '/home/runner/work/libvmm/libvmm/ci_build/examples/rust/make/qemu_virt_aarch64/release'
  make: *** [Makefile:38: /home/runner/work/libvmm/libvmm/ci_build/examples/rust/make/qemu_virt_aarch64/release/loader.img] Error 2
  make: Leaving directory '/home/runn
```

This is not really related to this PR but this repeatedly happens for the GitHub hosted runners when it tries to download the Linux images. I don't know if it's something to do with trustworthy.systems or GitHub's runners themselves. I have never seen these kind of timeout errors locally.

--- a/include/libvmm/virtio/virtio.h
+++ b/include/libvmm/virtio/virtio.h
@@ -9,11 +9,12 @@
 #include <libvmm/virtio/mmio.h>
 #include <libvmm/virtio/pci.h>
 
-/* The maximum size (number of elements) of a virtqueue. It is set
-* to 128 because I copied it from the camkes virtio device. If you find
-* out that the virtqueue gets full easily, increase the number.
-*/
-#define VIRTIO_QUEUE_SIZE 128
+/*
+ * Default maximum capacity of each virtIO queue. Currently applies to all
+ * virtIO queues for all virtIO devices. In the future, this could be
+ * configurable but our use-cases have not required it yet.
+ */
+#define VIRTIO_DEFAULT_QUEUE_SIZE 128
 
 /*
  * All terminology used and functionality of the virtIO device implementation

--- a/src/virtio/mmio.c
+++ b/src/virtio/mmio.c
@@ -106,7 +106,7 @@ static bool handle_virtio_mmio_reg_read(virtio_device_t *dev, size_t vcpu_id, si
         success = dev->funs->get_device_features(dev, &reg);
         break;
     case REG_RANGE(REG_VIRTIO_MMIO_QUEUE_NUM_MAX, REG_VIRTIO_MMIO_QUEUE_NUM):
-        reg = VIRTIO_QUEUE_SIZE;
+        reg = VIRTIO_DEFAULT_QUEUE_SIZE;
         break;
     case REG_RANGE(REG_VIRTIO_MMIO_QUEUE_READY, REG_VIRTIO_MMIO_QUEUE_NOTIFY):
         if (dev->regs.QueueSel < dev->num_vqs) {

--- a/src/virtio/pci.c
+++ b/src/virtio/pci.c
@@ -258,7 +258,7 @@ static bool virtio_pci_common_reg_read(virtio_device_t *dev, size_t vcpu_id, siz
         reg = dev->regs.ConfigGeneration;
         break;
     case REG_RANGE(VIRTIO_PCI_COMMON_Q_SIZE, VIRTIO_PCI_COMMON_Q_ENABLE):
-        reg = VIRTIO_QUEUE_SIZE;
+        reg = VIRTIO_DEFAULT_QUEUE_SIZE;
         break;
     case REG_RANGE(VIRTIO_PCI_COMMON_Q_ENABLE, VIRTIO_PCI_COMMON_Q_NOTIF_OFF):
         reg = dev->vqs[dev->regs.QueueSel].ready;

